server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/model/MailboxFactory.scala

```diff
@@ -105,21 +101,25 @@ class MailboxFactory @Inject() (subscriptionManager: SubscriptionManager) {
       .subscriptions(mailboxSession)
       .contains(mailboxMetaData.getPath.getName))
 
-    SMono.fromPublisher(quotas)
-      .map(quotas => Mailbox(
-        id = id,
-        name = name,
-        parentId = parentId,
-        role = role,
-        sortOrder = sortOrder,
-        unreadEmails = unreadEmails,
-        totalEmails = totalEmails,
-        unreadThreads = unreadThreads,
-        totalThreads = totalThreads,
-        myRights = myRights,
-        namespace = namespace,
-        rights = rights,
-        quotas = quotas,
-        isSubscribed = retrieveIsSubscribed))
+    name match {
+      case Left(error) => SMono.just(Left(error))
```

Comment:
Here we loose the context of what the error `String` is. We need more context but as said above, it depends what this error means.
Just think that Left is here to carry your error type but it can be any type.
At some point it can be painful to pass the Either all along and you can decide to throw.

Keep in mind that `Either`s compose: if you call `map` it will run only if no error and if you call `flatMap` the process in it can return an `Either` itself that will be flattened. That's where Either starts to shine